rpc/ExperimentalConsumer.ts

```diff
@@ -0,0 +1,245 @@
+    return this.stateCall("Metadata_metadata", new Uint8Array())
+  }
+
+  blockHashPendings: Deferred<string>[] = []
+  blockHash(blockNumber?: number) {
+    if (typeof blockNumber === "number") return this.archiveConsumer.blockHash(blockNumber)
+    const pending = deferred<string>()
+    this.blockHashPendings.push(pending)
+    return pending
+  }
+
+  blockPendings: Deferred<known.SignedBlock>[] = []
+  block(blockHash?: string) {
+    if (blockHash) return this.archiveConsumer.block(blockHash)
+    const pending = deferred<known.SignedBlock>()
+    this.blockPendings.push(pending)
+    return pending
+  }
+  blockFlush(followId: string, blockHash: string) {
+    if (this.blockPendings.length) {
+      const controller = new AbortController()
+      // TODO: why not emitting?
+      this.subscription("chainHead_unstable_body", undefined, [followId, blockHash], (_result) => {
+        while (this.blockPendings.length) {
+          const blockPending = this.blockPendings.shift()!
+          blockPending.resolve(null!)
+        }
+      }, controller.signal)
+    }
+  }
+
+  extrinsicsPendings: Deferred<Uint8Array>[] = []
+  extrinsics(blockHash?: string) {
+    if (blockHash) return this.archiveConsumer.block(blockHash)
+    const pending = deferred<Uint8Array>()
+    this.extrinsicsPendings.push(pending)
+    return pending
+  }
+  extrinsicsFlush(followId: string, blockHash: string) {
+    if (this.extrinsicsPendings.length) {
+      const controller = new AbortController()
+      this.subscription<{ event: "done"; result: string }>(
+        "chainHead_unstable_body",
+        undefined,
+        [followId, blockHash],
+        (result) => {
+          while (this.extrinsicsPendings.length) {
+            const blockPending = this.extrinsicsPendings.shift()!
+            blockPending.resolve(hex.decode(result.result))
+          }
+        },
+        controller.signal,
+      )
+    }
+  }
+
+  keys(key: Uint8Array, limit: number, start?: Uint8Array, blockHash?: string) {
+    return this.archiveConsumer.keys(key, limit, start, blockHash)
+  }
+
+  valuesKeys: Record<string, true> = {}
+  valuesPendings: [keys: string[], pending: Deferred<Uint8Array[]>][] = []
+  values(keys: Uint8Array[], blockHash?: string) {
+    if (blockHash) return this.archiveConsumer.values(keys, blockHash)
+    const keysEncoded = keys.map((key) => {
+      const encoded = hex.encodePrefixed(key)
+      this.valuesKeys[encoded] = true
+      return encoded
+    })
+    const pending = deferred<Uint8Array[]>()
+    this.valuesPendings.push([keysEncoded, pending])
+    return pending
+  }
+  valuesFlush(followId: string, blockHash: string) {
+    const keys = Object.entries(this.valuesKeys)
+    if (keys.length) {
+      const items = keys.map((key) => ({ key, type: "value" }))
+      const controller = new AbortController()
+      this.subscription<{ event: "items"; items: { key: string; value: string }[] }>(
```

Comment:
there are other events that we need to think how to handle
- `waiting-for-continue`, tricky because `.valuesPending` can be resolved only once
- `inaccessible`, retry a few times and reject?
- `error`, reject
- `disjoint`, reject